src/modules/timeseries.ts

```diff
@@ -36,6 +37,14 @@ export function fetchTimeseries(assetId: number) {
     dispatch({ type: SET_TIMESERIES, payload: { items: results.items } });
   };
 }
+// Functions
+export function fetchAndSetTimeseries(timeseriesId: number, redirect = false) {
+  return async (dispatch: ThunkDispatch<any, any, AnyAction>) => {
+    const results = await sdk.timeseries.retrieve([{ id: timeseriesId }]);
+    dispatch({ type: SET_TIMESERIES, payload: { items: results } });
+    dispatch(setTimeseriesId(timeseriesId, redirect));
+  };
+}
 let searchTimeseriesRID = 0;
```

Comment:
Not in this PR, but what is `RID`?

Reply:
lmao omg thats supposed to be searchTimeseriesR(andom)ID

because i wanted to make sure that the results always take the latest fetch (async nature means we needed a "ID"

then i realized the counter worked, so smh at the "randomness"

Reply:
Could you fix it in a PR? :D